components/Pagination/Pagination.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { number, string } from 'prop-types';
+import classNames from 'classnames';
+import LeftAngleIcon from 'static/images/icons/FontAwesome/angle-left-solid.svg';
+import RightAngleIcon from 'static/images/icons/FontAwesome/angle-right-solid.svg';
+import PaginationItem from './PaginationItem/PaginationItem';
+import styles from './Pagination.css';
+
+Pagination.propTypes = {
+  className: string,
+  currentPage: number.isRequired,
+  totalPages: number.isRequired,
+};
+
+Pagination.defaultProps = {
+  className: undefined,
+};
+
+const getPagination = (currentPage, totalPages) => {
+  // maximum length of the Pagination Bar, should be an odd integer, delault is 11
+  const MAX_VISIBLE_ELEMENTS = 11;
```

Comment:
Didn't mean to make you create a new variable! I was tryna get you to drop `ELEMENTS_ON_ONE_SIDE` and use `const MAX_VISIBLE_ELEMENTS = 7`

Reply:
I am actually not sure how to go about it. Are we trying to decrease the length of Pagination from `11` to `7` and drop an unnecessary variable?

Reply:
Something like
```javascript
const MAX_VISIBLE_ELEMENTS = 7;

  const isTruncatingRequired = totalPages >= MAX_VISIBLE_ELEMENTS;

  const isLeftSideLengthy = currentPage - 1 > (MAX_VISIBLE_ELEMENTS - 1) / 2;
  const isRightSideLengthy = totalPages - currentPage > (MAX_VISIBLE_ELEMENTS - 1) / 2;

  const shouldTruncateStart = isTruncatingRequired && isLeftSideLengthy;
  const shouldTruncateEnd = isTruncatingRequired && isRightSideLengthy;

  const truncateStartOnly = shouldTruncateStart && !shouldTruncateEnd;
  const truncateEndOnly = !shouldTruncateStart && shouldTruncateEnd;

  let paginationStart;
  let paginationEnd;

  if (truncateStartOnly) {
    paginationStart = totalPages - MAX_VISIBLE_ELEMENTS + 3;
    paginationEnd = totalPages;
  } else if (truncateEndOnly) {
    paginationStart = 1;
    paginationEnd = MAX_VISIBLE_ELEMENTS - 2;
  } else {
    paginationStart = shouldTruncateStart ? currentPage - (MAX_VISIBLE_ELEMENTS - 5) / 2 : 1;
    paginationEnd = shouldTruncateEnd ? currentPage + (MAX_VISIBLE_ELEMENTS - 5) / 2 : totalPages;
  }
```

Reply:
Ah whoops. I didn't realize `MAX_VISIBLE_ELEMENTS` was being used outside of defining `ELEMENTS_ON_ONE_SIDE`

Reply:
Yeah actually both variables are used as there are lot of edge cases to take care of.
> Nothing to do here then!

Okay cool.